### PR TITLE
Nucleo h745zi q serial console pins using dt

### PR DESCRIPTION
**Pull request by KozhinovAlexander**
Mostly copy of already merged PR #28849. The main changes proposed for other board with two serial consoles for m4 and m7 core projects.

P.S.: `hello_world` sample were tested by me and it works.

### CODE REVIEW

**Review by gmarull — COMMENTED** [review]
I think it's better to just squash into a single commit referring to the board

**Comment by Nukersson** [comment]
> I think it's better to just squash into a single commit referring to the board

done

File: boards/arm/nucleo_h745zi_q/nucleo_h745zi_q.dtsi [hunk 1]
```diff
@@ -4,6 +4,7 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#include <st/h7/stm32h743zitx-pinctrl.dtsi>
```
Comment (gmarull) [inline boards/arm/nucleo_h745zi_q/nucleo_h745zi_q.dtsi:7]: The pinctrl file does not point to the 745 SoC.

**Review by gmarull — CHANGES_REQUESTED** [review]

**Comment by lochej** [comment]
Thanks for your contribution 👍 
A new board for #28934

File: boards/arm/nucleo_h745zi_q/nucleo_h745zi_q.dtsi (hunk 1 above)
Comment (Nukersson) [reply boards/arm/nucleo_h745zi_q/nucleo_h745zi_q.dtsi]: You're absolutely right. It is fixed now (see last push). I have tested it and `hello_world` sample still works properly.

**Review by gmarull — APPROVED** [review]

**Review by lochej — APPROVED** [review]

**Comment by Nukersson** [comment]
@gmarull Sorry, just overseen your approve from just now. :) Pleas press approve button again 😄

**Review by galak — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Pull request merged**